File: nim/darts/Iteration2/darts.nim
import std/math
  
proc distanceFromCentre(x, y: float): float =
  hypot(x, y)
  
proc score*(x, y: float): int =
  let dist = distanceFromCentre(x, y);

  if (dist >= 0.0 and dist <= 1.0):
    return 10
  elif (dist >= 1.0 and dist <= 5.0):
    return 5
  elif (dist >= 5.0 and dist <= 10.0):
    return 1
  else:
    return 0

File: c/darts/Iteration2/darts.c
#include "darts.h"
#include <math.h>
#include <stdio.h>

const uint8_t CENTRE = 0;
const uint8_t INNER_RADIUS = 1;
const uint8_t MIDDLE_RADIUS = 5;
const uint8_t OUTER_RADIUS = 10;

static float distance_from_centre(coordinate_t coord) {
    return sqrt(pow(coord.x, 2) + pow(coord.y, 2));
}

uint8_t score(coordinate_t coord) {
    float dist = distance_from_centre(coord);
    
    if (dist >= CENTRE && dist <= INNER_RADIUS) return 10;
    if (dist >= INNER_RADIUS && dist <= MIDDLE_RADIUS) return 5;
    if (dist >= MIDDLE_RADIUS && dist <= OUTER_RADIUS) return 1;
    
    return 0;
}
